### ISSUE TITLE
Installation using Docker file- Getting Internal server error

### ISSUE THREAD

**Issue by AnithaVedhan** [mediagis/nominatim-docker#162]
HI,

Using attached docker file to build image and used below command to run
docker run --restart=always -p 6413:5432 -p 8000:8080 -d --name maldives4 devnominatimdb2:v1.1 bash /app/start.sh

Getting internal server error. PLease help

docker logs 546af7deab3ebce8f1b0d15036dce88234ecd78dfc870fa605ade96e             68f4365a
* Starting PostgreSQL 11 database server
   ...done.
* Starting Apache httpd web server apache2
AH00558: apache2: Could not reliably determine the server's fully qualified domain name, using 172.17.0.6. Set the 'ServerName' directive globally to suppress this message
*
2020-12-08 14:10:09.975 UTC [21] LOG:  listening on IPv4 address "0.0.0.0", port 5432
2020-12-08 14:10:09.975 UTC [21] LOG:  listening on IPv6 address "::", port 5432
2020-12-08 14:10:09.980 UTC [21] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
2020-12-08 14:10:09.995 UTC [22] LOG:  database system was shut down at 2020-12-04 16:55:18 UTC
2020-12-08 14:10:10.005 UTC [21] LOG:  database system is ready to accept connections
2020-12-08 14:10:10.559 UTC [29] [unknown]@[unknown] LOG:  incomplete startup packet

[Dockerfile.txt](https://github.com/mediagis/nominatim-docker/files/5680192/Dockerfile.txt)
![InternalServerError](https://user-images.githubusercontent.com/5010752/101925939-7ffac080-3bf8-11eb-819d-025f1f004676.png)

Thanks in advance!!

**Comment by AnithaVedhan** [comment]
CAn you please help with this issue?

**Comment by philipkozeny** [comment]
I'll try to take a look today. Is there any specific reason you are using 3.4 and not 3.5? Have you changed something in the Dockerfile from the repo?

**Comment by philipkozeny** [comment]
How did you run the init.sh command to initialise the database? Should be something like `docker run -t -v /home/me/nominatimdata:/data nominatim  sh /app/init.sh /data/<your_country>.osm.pbf postgresdata 4`

In the original `init.sh` we create the `www-data` user 

https://github.com/mediagis/nominatim-docker/blob/a2cd99637d718485bab376592f624af2355849ce/3.4/init.sh#L14

your error message suggest that the user does not exist. Do you have any debug output from the init step?

**Comment by AnithaVedhan** [comment]
Hi Philip,

In my docker file attached in the ticket, using below command RUN sh /app/init.sh to initialize database, so as part of the image this database will be initialized.

RUN cp -r ./code/3.4/start.sh /app/start.sh \
&&  cp -r ./code/3.4/startapache.sh /app/startapache.sh \
&&  cp -r ./code/3.4/startpostgres.sh /app/startpostgres.sh \
&&  cp -r ./code/3.4/init.sh /app/init.sh \ 
**&&  sh /app/init.sh /data/maldives-latest.osm.pbf postgresdata \**
&&  cp -r /data/postgresdata /var/lib/postgresql/11/main

Hope this can be done?

**Comment by philipkozeny** [comment]
The run command just copies the files into the docker image. Before starting the image with

`docker run --restart=always -p 6413:5432 -p 8000:8080 -d --name maldives4 devnominatimdb2:v1.1 bash /app/start.sh`

did you also do trigger the command to setup your database, users etc. with:

`docker run -t -v /home/me/nominatimdata:/data nominatim sh /app/init.sh /data/<your_country>.osm.pbf postgresdata 4`

?

**Comment by AnithaVedhan** [comment]
Hi Philip,

I have taken docker file from 3.5 and added steps to initialize database to the docker file, can you please validate this docker file as well?


Getting below error with 3.5 and postgres 12: 

Success. You can now start the database server using:

    /usr/lib/postgresql/12/bin/pg_ctl -D /data/postgresdata -l logfile start

initdb: warning: enabling "trust" authentication for local connections
You can change this by editing pg_hba.conf or using the option -A, or
--auth-local and --auth-host, the next time you run initdb.
waiting for server to start....2020-12-17 13:21:46.158 UTC [28] LOG:  starting PostgreSQL 12.5 (Ubuntu 12.5-0ubuntu0.20.04.1) on x86_64-pc-linux-gnu, compiled by gcc (Ubuntu 9.3.0-17ubuntu1~20.04) 9.3.0, 64-bit
2020-12-17 13:21:46.159 UTC [28] LOG:  liste
[Dockerfile.txt](https://github.com/mediagis/nominatim-docker/files/5709822/Dockerfile.txt)


ning on IPv4 address "127.0.0.1", port 5432
2020-12-17 13:21:46.159 UTC [28] LOG:  could not bind IPv6 address "::1": Cannot assign requested address
2020-12-17 13:21:46.159 UTC [28] HINT:  Is another postmaster already running on port 5432? If not, wait a few seconds and retry.
2020-12-17 13:21:46.161 UTC [28] FATAL:  could not create lock file "/var/run/postgresql/.s.PGSQL.5432.lock": Permission denied
2020-12-17 13:21:46.162 UTC [28] LOG:  database system is shut down
pg_ctl: could not start server
Examine the log output.
 stopped waiting
createuser: error: could not connect to database template1: could not connect to server: No such file or directory
        Is the server running locally and accepting
        connections on Unix domain socket "/var/run/postgresql/.s.PGSQL.5432"?
createuser: error: could not connect to database template1: could not connect to server: No such file or directory
        Is the server running locally and accepting
        connections on Unix domain socket "/var/run/postgresql/.s.PGSQL.5432"?
The command '/bin/sh -c sh /app/init.sh  /data/maldives-latest.osm.pbf postgresdata 4' returned a non-zero code: 1

**Comment by philipkozeny** [comment]
Can you also post the specific docker commands you have used to build / init the data / run the container?

**Comment by AnithaVedhan** [comment]
step1 :-using below command to build the image 
1)docker build -t "devnominatimdb5:v1.1" .

Step2:Then run below command to run the container

2)docker run --restart=always -p 6413:5432 -p 8000:8080 -d --name maldives devnominatimdb5:v1.1 bash /app/start.sh

**Comment by philipkozeny** [comment]
As stated in the Readme and In my comments you need to 

1) build the image with `docker build`
2) initialise the users, data and most importantly import the data into the db with `docker run .... /app/init.sh ...`
3) AFTER that you can start the command with `docker ... /app/start.sh`

Your error shows that you skipped step 2, you need to do that separately, since this commands checks if a DB exists, deletes it and then imports the data. Importing the data can take up days, so doing it on every start would defeat the purpose.

**Comment by AnithaVedhan** [comment]
ok Thanks Philip, will try now and update you.

**Comment by AnithaVedhan** [comment]
docker run -t -v /home/me/nominatimdata:/data nominatim sh /app/init.sh /data/<your_country>.osm.pbf postgresdata 4

in the above command how to refer image, that iam building, ideally database setup  and adding users should be done in image right?

**Comment by philipkozeny** [comment]
`/home/me/nominatimdata:/data **nominatim** sh`

bold marked "nominatim" is the image name. You can refer your own image there. Database setup probably, import should not be done in the image but in a mounted volume (as described in the readme)

**Comment by VShkaberda** [comment]
I have a similar issue during initialization. I tried to switch to Nominatim 3.5 and initialization (`docker run -t -v /home/osm/OSM_data/nominatim-data:/data nominatim  sh /app/init.sh /data/ukraine-latest.osm.pbf postgresdata 8`) returns error:

> 2020-12-30 21:14:57.261 UTC [29] LOG:  listening on IPv4 address "127.0.0.1", port 5432
2020-12-30 21:14:57.261 UTC [29] LOG:  could not bind IPv6 address "::1": Cannot assign requested address
2020-12-30 21:14:57.261 UTC [29] HINT:  Is another postmaster already running on port 5432? If not, wait a few seconds and retry.
2020-12-30 21:14:57.261 UTC [29] FATAL:  could not create lock file "/var/run/postgresql/.s.PGSQL.5432.lock": Permission denied
2020-12-30 21:14:57.261 UTC [29] LOG:  database system is shut down
 stopped waiting
pg_ctl: could not start server

After I switched back to Nominatim 3.4 initialization has been succeeded.
Maybe it is worth mentioning that I have this problem on CentOS-8 (4.18.0-193.19.1.el8_2.x86_64). On Ubuntu 18.04.4 LTS (GNU/Linux 4.15.0-91-generic x86_64) I had no problem and launched Nominatim 3.5 successfully.
